src/api/entities/SecurityToken/Settlements.ts

```diff
@@ -39,6 +39,8 @@ export class Settlements extends Namespace<SecurityToken> {
 
     const { from = await this.context.getCurrentIdentity(), to, amount } = args;
 
+    const { isDivisible } = await details();
```

Comment:
You should group both these awaits into a `Promise.all`

Reply:
but we are going to make two request every time although `from` is set

Reply:
```
const { to, amount } = args;
let { from } = args;
let isDivisible;

if (!from) {
  ([{ isDivisible }, from] = await Promise.all([details(), this.context.getCurrentIdentity()]);
} else {
  ({ isDivisible } = await details());
}
```